contrib/mypy/src/python/pants/contrib/mypy/mypy_task.py

```diff
@@ -149,21 +159,33 @@ def _collect_source_roots(self):
   def _interpreter_cache(self):
     return PythonInterpreterCache.global_instance()
 
+  @memoized_property
+  def _mypy_subsystem(self):
+    return MyPy.global_instance()
+
   def _get_mypy_pex(self, py3_interpreter: PythonInterpreter, *extra_pexes: PEX) -> PEX:
-    options = self.get_options()
-    mypy_version = options.version
+
+    def get_mypy_version() -> str:
+      task_version_configured = not self.get_options().is_default('version')
+      subsystem_version_configured = not self._mypy_subsystem.get_options().is_default('version')
+      if task_version_configured and subsystem_version_configured:
+        raise ValueError(
+          "Conflicting options for the MyPy version used. You used the new, preferred "
+          "`--mypy-version`, but also used the deprecated `--lint-mypy-version`.\nPlease use "
+          "only one of these (preferably `--mypy-version`)."
```

Comment:
You can be emphatic here: "Please use only --mypy-version." This is not preferable, this is mandatory...

Reply:
Well, technically it’s possible to use either for two more releases. They’ll still get the deprecation warning if they use the task level option. 

Feel free to push up the change, though. I’m afk the rest of the day (going to the sloth sanctuary!)